### PR TITLE
Better testing for Monge-Ampere boundary conditions

### LINKED ISSUE #79: Better testing for Monge-Ampere boundary conditions
There currently isn't coverage for all the different cases, in particular non-axis-aligned boundaries.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #79.
Partially addresses #19.

This PR achieves 100% code coverage for the `monge_ampere` module. In particular, it introduces unit tests for the boundary condition logic for the equidistribution equation. The boundary condition code is refactored, including pulling it out as a separate method so it can be more easily targeted.

The PR also introduces a new `math` module including functions for generating equations of lines and planes. These are used to verify that boundary conditions are only applied on straight line boundary segments (2D) and planar boundary segments (3D). The new module has 100% code coverage.

Finally, an additional error check test is added for the `tangling` module to get that to 100% code coverage, too.

### CODE REVIEW

File: movement/monge_ampere.py
Comment (jwallwork23) [inline movement/monge_ampere.py:147]: This was needed because libsupermesh is only implemented in 2D and 3D.

**Comment by jwallwork23** [comment]
Putting down everyone as reviewer in case they have time to look at this. Would be good to get it merged so we can carry on with other MA changes. Feel free to only leave a partial review with comments on certain parts if you like.

**Comment by ddundo** [comment]
Thanks @jwallwork23! I have yet to dive deep into the underlying movement code so I will leave it to others to review for now :) otherwise I will get into it in the next 2-3 weeks

**Comment by jwallwork23** [comment]
@stephankramer Do you have time to review this by any chance? Would be much appreciated.

File: movement/monge_ampere.py
Comment (ddundo) [inline movement/monge_ampere.py:218]: Just for my understanding: `ffacet_bc` says that there is no movement of nodes connecting two different facets? Maybe this could benefit from an explicit comment - and then relating it to tangential movement in `bc2` below :)

**Review by ddundo — APPROVED** [review]
Sorry for being slow with this @jwallwork23! I still don't feel 100% comfortable reviewing this, but I think it looks good! Left one potentially important comment and the rest are minor :)

File: movement/math.py
Comment (ddundo) [reply movement/math.py:15]: Actually I think it's great as it is. Sorry! Will resolve :)

File: movement/monge_ampere.py
Comment (jwallwork23) [reply movement/monge_ampere.py:199]: Yes but it's already set a few lines above.

File: movement/monge_ampere.py
Comment (jwallwork23) [reply movement/monge_ampere.py]: Great point - fixed in c02feef.

File: test/test_monge_ampere.py
Comment (jwallwork23) [reply test/test_monge_ampere.py]: Addressed in a489a7d.

File: movement/monge_ampere.py
Comment (jwallwork23) [reply movement/monge_ampere.py:264]: Good point. Hopefully 8112172 makes it clearer.

**Comment by jwallwork23** [comment]
Note that #115 will need merging into the branch associated with this PR before we merge this one.

File: movement/monge_ampere.py
Comment (ddundo) [reply movement/monge_ampere.py:243]: Ah sorry, I missed that!

**Pull request merged**
